cpp/src/io/utilities/config_utils.hpp

```diff
@@ -30,6 +30,41 @@ inline std::string getenv_or(std::string const& env_var_name, std::string_view d
   return std::string{(env_val == nullptr) ? default_val : env_val};
 }
 
+namespace cufile_integration {
+
+namespace {
+/**
+ * @brief Defines which cuFile usage to enable.
+ */
+enum class usage_policy : uint8_t { OFF, GDS, ALWAYS };
+
+/**
+ * @brief Get the current usage policy.
+ */
+inline usage_policy get_env_policy()
+{
+  static auto const env_val = getenv_or("LIBCUDF_CUFILE_POLICY", "GDS");
+  if (env_val == "OFF") return usage_policy::OFF;
+  if (env_val == "ALWAYS") return usage_policy::ALWAYS;
+  return usage_policy::GDS;
```

Comment:
Since you're defaulting `env_val` to `GDS`, perhaps a switch statement would be more appropriate (maybe with a `__builtin_unreachable` at the end)? With the current approach you maintain the default in two places, the `getenv_or` call and the default return.

Reply:
changed to throw for invalid values